### PR TITLE
[java] Update rule LogicInversion

### PR DESCRIPTION
**Pull request by adangel**
Part of #2701

### CODE REVIEW

**Comment by pmd-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to pmd/7.0.x:<br />
This changeset changes 0 violations,<br />
introduces 25 new violations, 0 new errors and 0 new configuration errors,<br />
removes 0 violations, 0 errors and 0 configuration errors.<br />
<a href="https://chunk.io/pmd/2545b3e98046487ba3846fda81f2eb32/diff1/index.html">Full report</a></td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="true">Compared to master:<br />
This changeset changes 31211 violations,<br />
introduces 21291 new violations, 1 new errors and 0 new configuration errors,<br />
removes 135620 violations, 8 errors and 3 configuration errors.<br />
<a href="https://chunk.io/pmd/2545b3e98046487ba3846fda81f2eb32/diff2/index.html">Full report</a></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: pmd-java/src/main/resources/category/java/design.xml [hunk 1]
```diff
@@ -927,7 +927,7 @@ Use opposite operator instead of negating the whole expression with a logic comp
             <property name="xpath">
                 <value>
 <![CDATA[
-//UnaryExpressionNotPlusMinus[@Image='!']/PrimaryExpression/PrimaryPrefix/Expression[EqualityExpression or RelationalExpression]
+//UnaryExpression[@Operator='!']/InfixExpression[@Operator = ('==', '<', '>', '<=', '>=')]
```
Comment (jsotuyod) [inline pmd-java/src/main/resources/category/java/design.xml:930]: shouldn't this also include `!=` testing for the double negative?
```suggestion
//UnaryExpression[@Operator='!']/InfixExpression[@Operator = ('==', '!=', '<', '>', '<=', '>=')]
```

**Comment by jsotuyod** [comment]
Really happy to see this rule is actually detecting 25 more violations, and all of them being right. Parenthesis seemed to be confusing the old rule, another big win for the updated grammar.

File: pmd-java/src/main/resources/category/java/design.xml (hunk 1 above)
Comment (adangel) [reply pmd-java/src/main/resources/category/java/design.xml:930]: Probably yes, then we would be able to detect code like this:

```java
if (!(a != b)) { ...}  // instead of "a == b"
```

I'll try to add a test case for this.

**Comment by adangel** [comment]
> Really happy to see this rule is actually detecting 25 more violations, and all of them being right. Parenthesis seemed to be confusing the old rule, another big win for the updated grammar.

Actually these 25 violation are new, because the rule was disabled till now :nerd_face:  . Comparing with master shows, that we detect exactly the same violations for this rule (no additions, no removals).
